dotCMS/src/main/java/com/dotcms/graphql/GraphQLCache.java

```diff
@@ -46,32 +45,19 @@
      */
 
     private Optional<String> get(final String key, final boolean refresh,
-            final Supplier<String> valueSupplier, Integer ttl) {
+            final Supplier<String> valueSupplier, final long ttl) {
         if(cannotCache()) {
             return Optional.empty();
         }
 
-        Optional<String> result = Optional.empty();
         final String cacheKey = hashKey(key);
-        final Tuple2<String, LocalDateTime> resultExpireTimeTuple
-                = (Tuple2<String, LocalDateTime>) cache.getNoThrow(cacheKey, getPrimaryGroup());
-
-        if(UtilMethods.isSet(resultExpireTimeTuple)) {
-            if(refresh) { // return from cache even if expired and refresh in background
-                result = Optional.of(resultExpireTimeTuple._1());
-                refreshKey(key, valueSupplier, ttl);
-            } else {
-                final LocalDateTime expireTime = resultExpireTimeTuple._2();
-
-                if (expireTime == null || expireTime.isAfter(LocalDateTime.now())) {
-                    result = Optional.of(resultExpireTimeTuple._1());
-                } else { // expired, let's remove from cache
-                    remove(key);
-                }
-            }
-        }
+        final Object value = cache.getNoThrow(cacheKey, getPrimaryGroup());
+
+        refreshKey(cacheKey, valueSupplier, ttl);
 
-        return result;
+        return value instanceof ExpirableCacheEntry
```

Comment:
shpuld it be Expirable? Also I think is better use the Expirable.class.isInstance()

Reply:
@freddyucv ExpirableCacheEntry is ok - Expirable doesn't have the "getResults" method, only the getTTL